x/tradeshield/keeper/pending_spot_order.go

```diff
@@ -320,6 +327,13 @@ func (k Keeper) ExecuteLimitBuyOrder(ctx sdk.Context, order types.SpotOrder) (*a
 		MinAmount: sdk.NewCoin(order.OrderTargetDenom, sdkmath.ZeroInt()),
 		MaxAmount: order.OrderAmount,
 	})
+	tolerance := res.SwapFee.Add(res.Slippage)
+	if res.WeightBonus.IsNegative() {
+		tolerance = tolerance.Add(res.WeightBonus.Abs())
+	}
```

Comment:
if weightBonus is positive, shouldn't it be subtracted?

Reply:
when it is -ve so it will be charged and subtracted from total amount, therefore added to tolerance